scipy-1.0/scipy-optimize.tex

```diff
@@ -21,66 +21,8 @@ \subsubsection*{Linear Optimization}
-the solution of large problems (e.g.~problems with more than 1000 variables).
-Other methods are appropriate for large-scale problems, including
-\texttt{CG}, which does not use second derivatives and thus has low memory requirements, and \texttt{L-BFGS-B}, which maintains a simple and compact representation of the Hessian.
-\texttt{Newton-CG}, \texttt{trust-ncg} and \texttt{trust-krylov} use second order derivatives to achieve faster
-convergence, but are still appropriate for large problems because they factorize the Hessian in an inexact and iterative way that does not require the Hessian to be stored.
-The method \texttt{trust-krylov}, introduced in the latest release, is a good compromise between
-\texttt{trust-ncg} and \texttt{trust-exact}: it provides a more accurate solution to the trust-region
-subproblem than \texttt{trust-ncg}, but it does not require the Hessian to be stored and factored as in \texttt{trust-exact}.
-
-While most \texttt{minimize} development prior to SciPy 1.0 has focused on unconstrained minimization, 
-% methods \texttt{SLSQP} and \texttt{COBYLA}\footnote{We note that any equality constraint $c_{\text{eq}}(x) = 0$ can be expressed as two inequality constraints $c_{\text{eq}}(x) \leq 0$ and $-c_{\text{eq}}(x) \leq 0$} 
-\texttt{SLSQP} supports fully general nonlinear programming. However, it does not provide special treatment for sparsity, limiting its utility to small problems. Accordingly, ongoing development efforts for \texttt{minimize} are focused on the addition of nonlinear programming algorithms for large, sparse problems. 
-
+The \texttt{minimize} function unifies several methods of finding local minima of nonlinear optimization problems in a common interface. Four new methods for unconstrained optimization were added to \texttt{minimize} in recent versions of SciPy: \texttt{dogleg}, \texttt{trust-ncg}, \texttt{trust-exact}, and \texttt{trust-krylov}. All are trust-region methods that build a local model of the objective function based on first- and second-derivative information, approximate the best point within a local ``trust region'', and iterate until a local minimum of the original objective function is reached, but each has unique characteristics that make it appropriate for certain types of problems. For instance, \texttt{trust-exact} achieves fast convergence by solving the trust-region subproblem almost exactly, but it requires the second-derivative Hessian matrix to be stored and factored every iteration, which may preclude the solution of large problems ($\geq 1000$ variables). On the other hand, \texttt{trust-ncg} and \texttt{trust-krylov} are well-suited to large-scale optimization problems because they do not need to store and factor the Hessian explicitly, instead using second-derivative information in a faster, approximate way. A detailed comparison of the characteristics of all \texttt{minimize} methods is presented in Table~\ref{tab:minimize-si}.
```

Comment:
```suggestion
The \texttt{minimize} function unifies,  in a common interface, several methods for finding local minima of nonlinear optimization problems. Four new methods for unconstrained optimization were added to \texttt{minimize} in recent versions of SciPy: \texttt{dogleg}, \texttt{trust-ncg}, \texttt{trust-exact}, and \texttt{trust-krylov}. All are trust-region methods that build a local model of the objective function based on first and second derivative information, approximate the best point within a local ``trust region'', and iterate until a local minimum of the original objective function is reached, but each has unique characteristics that make it appropriate for certain types of problems. For instance, \texttt{trust-exact} achieves fast convergence by solving the trust-region subproblem almost exactly, but it requires the second-derivative Hessian matrix to be stored and factored every iteration, which may preclude the solution of large problems ($\geq 1000$ variables). On the other hand, \texttt{trust-ncg} and \texttt{trust-krylov} are well-suited to large-scale optimization problems because they do not need to store and factor the Hessian explicitly, instead they use second derivative information in a faster, approximate way. A detailed comparison of the characteristics of all \texttt{minimize} methods is presented in Table~\ref{tab:minimize-si}.
```

Reply:
@apbard, I can't tell the difference between your suggestion and the original.

Reply:
@andyfaff I see...it is rendered as just one line. I'll fix it

Reply:
```diff
-The \texttt{minimize} function unifies several methods of finding local minima 
-of nonlinear optimization problems in a common interface.
+The \texttt{minimize} function unifies, in a common interface, several methods 
+for finding local minima of nonlinear optimization problems. 
 Four new methods for unconstrained optimization were added to \texttt{minimize} 
 in recent versions of SciPy: \texttt{dogleg}, \texttt{trust-ncg}, 
 \texttt{trust-exact}, and \texttt{trust-krylov}. 
 All are trust-region methods that build a local model 
-of the objective function based on first- and second-derivative information, 
+of the objective function based on first and second derivative information, 
 approximate the best point within a local ``trust region'', and iterate until 
 a local minimum of the original objective function is reached, but each has 
 unique characteristics that make it appropriate for certain types of problems. 
 For instance, \texttt{trust-exact} achieves fast convergence by solving 
 the trust-region subproblem almost exactly, but it requires the 
-second-derivative Hessian matrix to be stored and factored every iteration, 
+second derivative Hessian matrix to be stored and factored every iteration, 
 which may preclude the solution of large problems ($\geq 1000$ variables). 
 On the other hand, \texttt{trust-ncg} and \texttt{trust-krylov} are well-suited 
 to large-scale optimization problems because they do not need to store and factor 
-the Hessian explicitly, instead using second-derivative information in a faster, 
+the Hessian explicitly, instead they use second derivative information in a faster, 
 approximate way. A detailed comparison of the characteristics of 
 all \texttt{minimize} methods is presented in Table~\ref{tab:minimize-si}.
```

Reply:
Thanks for highlighting the changes.

I agree with removing the hyphens. Good catch.

Re: making "in a common interface" a parenthetical clause, I can see arguments both ways.
As the sentence was, "unifies several methods of finding local minima..." was unfortunately separated from "...in a common interface." I would prefer to be able to squash those together. 
Now "unifies, in a common interface, several methods" interrupts the sentence. Is it important enough to interject that thought?

I will fix the hyphens but leave the original "in a common interface", if you don't mind? Of course, if you think it's important enough. please open a PR for it.

Reply:
I also changed the `several methods of finding local` to  `several methods `**for**` finding local`, isn't this more correct?

Do you think something like `The \texttt{minimize} provides a common (or shared) interface for several methods of... ` solves the issue? If not, your version it is fine.

Reply:
Yes, both of these are good. I'll fix.

Reply:
I went with "unified" interface. Means the same thing but suggests that they might have been separate before, and I think that's why the original author wrote something like "unifies ... in a common interface."